### ISSUE TITLE
Multi-label image classification bug

### ISSUE THREAD

**Issue by hjw-0909** [open-mmlab/mmclassification#1439]
### Branch

master branch (0.24 or other 0.x version)

### Describe the bug
I want to implement the loss of the first 10 categories as CE loss, and use BCE loss for each of the following categories.
During training, it is said that my loss has always been nan
<img width="589" alt="image" src="https://user-images.githubusercontent.com/56182057/227752437-c7d5ec9e-a021-4f8c-a204-3d1566c9528d.png">
My label_gt is [1, 0, 0, 0, 1, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 1, 0, 0, 0]
My config file：
dataset_type = 'Multi_Lanes'
img_norm_cfg = dict(
    mean=[123.675, 116.28, 103.53], std=[58.395, 57.12, 57.375], to_rgb=True)
train_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(type='RandomResizedCrop_lanes', size=224),
    dict(type='RandomFlip', flip_prob=0.5, direction='horizontal'),
    dict(
        type='Normalize',
        mean=[123.675, 116.28, 103.53],
        std=[58.395, 57.12, 57.375],
        to_rgb=True),
    dict(type='ImageToTensor', keys=['img']),
    dict(type='ToTensor', keys=['gt_label']),
    dict(type='Collect', keys=['img', 'gt_label'])
]
test_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(type='Resize', size=(256, -1)),
    dict(type='CenterCrop', crop_size=224),
    dict(
        type='Normalize',
        mean=[123.675, 116.28, 103.53],
        std=[58.395, 57.12, 57.375],
        to_rgb=True),
    dict(type='ImageToTensor', keys=['img']),
    dict(type='Collect', keys=['img'])
]
data = dict(
    samples_per_gpu=2,
    workers_per_gpu=2,
    train=dict(
        type='Multi_Lanes',
        data_prefix='/mnt/cfs/mmclassification/data',
        ann_file='/mnt/cfs/mmclassification/labels.txt',
        pipeline=[
            dict(type='LoadImageFromFile'),
            dict(type='RandomResizedCrop_lanes', size=224),
            dict(type='RandomFlip', flip_prob=0.5, direction='horizontal'),
            dict(
                type='Normalize',
                mean=[123.675, 116.28, 103.53],
                std=[58.395, 57.12, 57.375],
                to_rgb=True),
            dict(type='ImageToTensor', keys=['img']),
            dict(type='ToTensor', keys=['gt_label']),
            dict(type='Collect', keys=['img', 'gt_label'])
        ]),
    val=dict(
        type='Multi_Lanes',
        data_prefix='/mnt/cfs/mmclassification/data',
        ann_file='/mnt/cfs/mmclassification/labels.txt',
        pipeline=[
            dict(type='LoadImageFromFile'),
            dict(type='Resize', size=(256, -1)),
            dict(type='CenterCrop', crop_size=224),
            dict(
                type='Normalize',
                mean=[123.675, 116.28, 103.53],
                std=[58.395, 57.12, 57.375],
                to_rgb=True),
            dict(type='ImageToTensor', keys=['img']),
            dict(type='Collect', keys=['img'])
        ]),
    test=dict(
        type='Multi_Lanes',
        data_prefix='/mnt/cfs/data/env_cls/val',
        ann_file='/mnt/cfs/data/env_cls/meta/val.txt',
        pipeline=[
            dict(type='LoadImageFromFile'),
            dict(type='Resize', size=(256, -1)),
            dict(type='CenterCrop', crop_size=224),
            dict(
                type='Normalize',
                mean=[123.675, 116.28, 103.53],
                std=[58.395, 57.12, 57.375],
                to_rgb=True),
            dict(type='ImageToTensor', keys=['img']),
            dict(type='Collect', keys=['img'])
        ]))
evaluation = dict(
    interval=1, metric=['mAP', 'CP', 'OP', 'CR', 'OR', 'CF1', 'OF1'])
optimizer = dict(type='SGD', lr=0.1, momentum=0.9, weight_decay=0.0001)
optimizer_config = dict(grad_clip=None)
lr_config = dict(policy='step', step=[30, 60, 90])
runner = dict(type='EpochBasedRunner', max_epochs=100)
checkpoint_config = dict(interval=1)
log_config = dict(interval=30, hooks=[dict(type='TextLoggerHook')])
dist_params = dict(backend='nccl')
log_level = 'INFO'
load_from = None
resume_from = None
workflow = [('train', 1)]
model = dict(
    type='ImageClassifier',
    backbone=dict(type='VGG', depth=16, num_classes=24),
    neck=None,
    head=dict(
        type='MultiLabelClsHead',
        loss=dict(type='CrossEntropyLoss', use_sigmoid=True, loss_weight=1.0)))
work_dir = './work_dirs/resnet_multi_lanes'
gpu_ids = [0]

### Environment

'TorchVision': '0.11.1',
 'OpenCV': '4.7.0',
 'MMCV': '1.7.1',
 'MMCV Compiler': 'GCC 7.3',
 'MMCV CUDA Compiler': '11.1',
 'MMClassification': '0.25.0+e206c82'}

**Comment by boxbox-0427** [comment]
Is there any tutorial about Multi-label using `mmpretrain` or `mmcls`?  Can't find any doc about it

**Comment by ccxi1008** [comment]
> ### Branch
> master branch (0.24 or other 0.x version)
> 
> ### Describe the bug
> I want to implement the loss of the first 10 categories as CE loss, and use BCE loss for each of the following categories. During training, it is said that my loss has always been nan <img alt="image" width="589" src="https://user-images.githubusercontent.com/56182057/227752437-c7d5ec9e-a021-4f8c-a204-3d1566c9528d.png"> My label_gt is [1, 0, 0, 0, 1, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 1, 0, 0, 0] My config file： dataset_type = 'Multi_Lanes' img_norm_cfg = dict( mean=[123.675, 116.28, 103.53], std=[58.395, 57.12, 57.375], to_rgb=True) train_pipeline = [ dict(type='LoadImageFromFile'), dict(type='RandomResizedCrop_lanes', size=224), dict(type='RandomFlip', flip_prob=0.5, direction='horizontal'), dict( type='Normalize', mean=[123.675, 116.28, 103.53], std=[58.395, 57.12, 57.375], to_rgb=True), dict(type='ImageToTensor', keys=['img']), dict(type='ToTensor', keys=['gt_label']), dict(type='Collect', keys=['img', 'gt_label']) ] test_pipeline = [ dict(type='LoadImageFromFile'), dict(type='Resize', size=(256, -1)), dict(type='CenterCrop', crop_size=224), dict( type='Normalize', mean=[123.675, 116.28, 103.53], std=[58.395, 57.12, 57.375], to_rgb=True), dict(type='ImageToTensor', keys=['img']), dict(type='Collect', keys=['img']) ] data = dict( samples_per_gpu=2, workers_per_gpu=2, train=dict( type='Multi_Lanes', data_prefix='/mnt/cfs/mmclassification/data', ann_file='/mnt/cfs/mmclassification/labels.txt', pipeline=[ dict(type='LoadImageFromFile'), dict(type='RandomResizedCrop_lanes', size=224), dict(type='RandomFlip', flip_prob=0.5, direction='horizontal'), dict( type='Normalize', mean=[123.675, 116.28, 103.53], std=[58.395, 57.12, 57.375], to_rgb=True), dict(type='ImageToTensor', keys=['img']), dict(type='ToTensor', keys=['gt_label']), dict(type='Collect', keys=['img', 'gt_label']) ]), val=dict( type='Multi_Lanes', data_prefix='/mnt/cfs/mmclassification/data', ann_file='/mnt/cfs/mmclassification/labels.txt', pipeline=[ dict(type='LoadImageFromFile'), dict(type='Resize', size=(256, -1)), dict(type='CenterCrop', crop_size=224), dict( type='Normalize', mean=[123.675, 116.28, 103.53], std=[58.395, 57.12, 57.375], to_rgb=True), dict(type='ImageToTensor', keys=['img']), dict(type='Collect', keys=['img']) ]), test=dict( type='Multi_Lanes', data_prefix='/mnt/cfs/data/env_cls/val', ann_file='/mnt/cfs/data/env_cls/meta/val.txt', pipeline=[ dict(type='LoadImageFromFile'), dict(type='Resize', size=(256, -1)), dict(type='CenterCrop', crop_size=224), dict( type='Normalize', mean=[123.675, 116.28, 103.53], std=[58.395, 57.12, 57.375], to_rgb=True), dict(type='ImageToTensor', keys=['img']), dict(type='Collect', keys=['img']) ])) evaluation = dict( interval=1, metric=['mAP', 'CP', 'OP', 'CR', 'OR', 'CF1', 'OF1']) optimizer = dict(type='SGD', lr=0.1, momentum=0.9, weight_decay=0.0001) optimizer_config = dict(grad_clip=None) lr_config = dict(policy='step', step=[30, 60, 90]) runner = dict(type='EpochBasedRunner', max_epochs=100) checkpoint_config = dict(interval=1) log_config = dict(interval=30, hooks=[dict(type='TextLoggerHook')]) dist_params = dict(backend='nccl') log_level = 'INFO' load_from = None resume_from = None workflow = [('train', 1)] model = dict( type='ImageClassifier', backbone=dict(type='VGG', depth=16, num_classes=24), neck=None, head=dict( type='MultiLabelClsHead', loss=dict(type='CrossEntropyLoss', use_sigmoid=True, loss_weight=1.0))) work_dir = './work_dirs/resnet_multi_lanes' gpu_ids = [0]
> 
> ### Environment
> 'TorchVision': '0.11.1', 'OpenCV': '4.7.0', 'MMCV': '1.7.1', 'MMCV Compiler': 'GCC 7.3', 'MMCV CUDA Compiler': '11.1', 'MMClassification': '0.25.0+e206c82'}
> 
> ### Other information
> _No response
Same question. Can I ask you how to build the data folder and dataset config? I can't find any guide about multilabel training. I only fine a model template like this. Don't know how to use it .

<img width="1410" height="756" alt="Image" src="https://github.com/user-attachments/assets/7a2268ff-9aa1-4bde-a743-612f57a44a72" />
